newjersey/navigator.business.nj.gov
# feat: [#185151322] Post-onboarding Questions dropdown support

## Description

These changes allow for post-onboarding questions to have their own CTAs, including a button-based CTA and a dropdown-based CTA, as per ticket. This flexibility is demonstrated with the construction renovation post-onboarding question, which is now configured with a dropdown with the two provided links. The functionality is also mirrored in the CMS, giving CMS users immediate feedback to see how their CTA will appear.

In order for a dropdown to appear, all five greymatter attributes must be defined:

- dropdown text (the text that should appear inside the dropdown)
- link 1 name and destination
- link 2 name and destination

If only the link 1 name and destination (or link 2 name and destination) are specified, a CTA button will appear instead. If both are provided, nothing will appear until the dropdown text is provided.

### Ticket

[#185151322](https://www.pivotaltracker.com/story/show/185151322)

### Approach

No dependencies or migration are necessary for acceptance and merger of this pull request.

### Steps to Test

1. Log in to the CMS to make changes.
2. Navigate in the CMS to the post-onboarding questions section (🟨 MISC -> Post Onboarding Content)
3. Click the `construction-renovation` post-onboarding question.
4. Observe changes to the behavior of the gray footer section of the post-onboarding question preview on the right side as the relevant items (dropdown name, link 1 name and destination, link 2 name and destination) on the left side are changed.

### Notes

Thanks to @aloverso and @FarazA22 for assistance in getting this over the line.

## Code author checklist
